lib/threelang/web3gw/money.v

```
@@ -0,0 +1,169 @@
+}
+
+fn swap(mut h Web3GWHandler, action Action) ! {
+	from := action.params.get('from')!
+	to := action.params.get('to')!
+	amount := action.params.get('amount')!
+
+	if from == 'eth' && to == 'tft' {
+		res := h.clients.eth_client.swap_eth_for_tft(amount)!
+		h.logger.info(res)
+	} else if from == 'tft' && to == 'eth' {
+		res := h.clients.eth_client.swap_tft_for_eth(amount)!
+		h.logger.info(res)
+	} else if from == 'tft' && to == 'xlm' {
+		res := h.clients.str_client.swap(
+			amount: amount
+			source_asset: from
+			destination_asset: to
+		)!
+		h.logger.info(res)
+	} else if from == 'xlm' && to == 'tft' {
+		res := h.clients.str_client.swap(
+			amount: amount
+			source_asset: from
+			destination_asset: to
+		)!
+		h.logger.info(res)
+	} else {
+		return error('unsupported swap')
+	}
+}
+
+fn balance(mut h Web3GWHandler, action Action) ! {
+	channel := action.params.get('channel')!
+	mut currency := action.params.get_default('currency', '')!
+
+	if currency == '' {
+		currency = web3gw.default_currencies[channel]!
+	}
+
+	if channel == 'bitcoin' {
+		account := action.params.get('account')!
+		res := h.clients.btc_client.get_balance(account)!
+		h.logger.info('balance on ${channel} is ${res}')
+	} else if channel == 'ethereum' && currency == 'eth' {
+		address := h.clients.eth_client.address()!
+		res := h.clients.eth_client.balance(address)!
+		h.logger.info('balance on ${channel} is ${res}')
+	} else if channel == 'ethereum' && currency == 'tft' {
+		res := h.clients.eth_client.tft_balance()!
+		h.logger.info('balance on ${channel} is ${res}')
+	} else if channel == 'stellar' {
+		address := h.clients.str_client.address()!
+		res := h.clients.str_client.balance(address)!
+		h.logger.info('balance on ${channel} is ${res}')
+	} else if channel == 'tfchain' {
+		address := h.clients.tfc_client.address()!
+		res := h.clients.tfc_client.balance(address)!
+		h.logger.info('balance on ${channel} is ${res}')
+	} else {
+		return error('unsupported balance')
```

Maybe return the error "Unsupported channeld. Should be one of: (list them here)"